specs/capri/orchestration_automation_design.md

```diff
@@ -0,0 +1,523 @@
+User can write a YAML or JSON template to register a service catalog which represents a specific user’s business process. `Service`, `Workflow` and `Task` can be defined inside it. A Service contains a couple of workflows, each workflow consists of multiple tasks. Every task can have one or more than one action.
+
+* **Connection Manager**: Connection Manager handles the workflow connector. It manages the connection between the OpenSDS Service Manager and the workflow manager. There can be different adapters which acts as workflow connector for OpenSDS workflow orchestration. Connector acts as an OpenSDS orchestration adapter for inbound and outbound integration with the Workflow Manager. In the current implementation, we use Stackstorm as the Workflow Manager and connector connects StackStorm to handle the interaction between service catalog manager and StackStorm.
+
+**StackStorm**: It’s an `IFTTT` framework which offers advanced automation, letting users create complex workflows to automate your world. Stackstorm breaks automation down into discrete actions and ties these actions together into workflows that can perform complex operations. StackStorm also provides a sophisticated event bus that can be used to implement event-based automation. Stackstorm enables OpenSDS Orchestration and Automation for the generic workflow actions based on OpenSDS projects such as hotpot, gelato and so forth.
+
+###Terminology
+
+ Service Catalog Manager
+It manages service catalogs and is responsible for maintaining the Service Catalog, ensuring that all information within the Service Catalog is accurate and up-to-date.
+
+ Service
+It represents user defined business process. User can write a YAML or JSON template to register a service which on validation registers a workflow. Also, it can be pre-defined workflows shipped with the product.
+
+ Instance
+It is a consumable service. Once the registration is done, user can consume the service.
+
+ Execution
+A service instance can be executed immediately or scheduled later. It helps in tracking execution progress, result of service instance.
+
+ Connection Manager
+Connection Manager handles the workflow connectors. It manages the connection between the OpenSDS Service Manager and the workflow manager.
+
+ Connector
+It is the OpenSDS orchestration adapter for inbound and outbound integration with the Workflow Manager.  
+
+ Workflow Manager
+It is the external framework to organize, centralize, and standardize workflows. 
+
+ Workflow
+It stitches actions together, define the order, transition conditions, and passing the data. 
+
+ Actions
+Actions are pieces of code that can perform arbitrary automation tasks.
+
+## Design details
+![Orchestration Design Diagram](Orchestration_and_Automation_Design.PNG?raw=true "Orchestration Design Diagram")
```

Comment:
Something is not right.  I cannot view this diagram when I view the doc.

Reply:
@xing-yang file extension needs change to .png
We will update the design docs for other contents like API, sequence diagram.